scripts/zkai_native_seq32_attention_mlp_expanded_label_probe_gate.py

```
@@ -0,0 +1,628 @@
+    except json.JSONDecodeError as err:
+        raise ExpandedLabelProbeGateError(f"{label} is not valid JSON: {err}") from err
+    if not isinstance(value, dict):
+        raise ExpandedLabelProbeGateError(f"{label} must be a JSON object")
+    return value, raw
+
+
+def source_artifact(path: pathlib.Path, artifact_id: str, expected_sha: str) -> dict[str, Any]:
+    raw = read_bounded_repo_file(path, artifact_id, MAX_SOURCE_ARTIFACT_BYTES)
+    digest = sha256_bytes(raw)
+    if digest != expected_sha:
+        raise ExpandedLabelProbeGateError(f"{artifact_id} source digest drift")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)